Delta Lake diff: Add documentation for the Delta Lake diff plugin (#5506)
treeverse/lakeFS: docs/integrations/delta.md

{% include toc.html %}

## Beta: Viewing Delta Lake table changes in lakeFS 

> ### Enable Delta Lake tables diff

Reviewer: ```suggestion
### Installing the plugin
```

> ### Enable the Delta Lake tables diff

Reviewer 2: Nit:
```suggestion
### Enable Delta Lake tables diff
```

> The Delta Lake diff functionality is enabled using a binary plugin. This binary plugin can be found as part of the latest lakeFS
> [release](https://github.com/treeverse/lakeFS/releases/latest) (starting from 0.100.0), and can also be built locally using the `make delta-plugin` command
> (run from lakeFS's repository root). After getting it, place it under `~/.lakefs/plugins/diff`.  
> lakeFS will look for a binary named `delta_diff` under `~/.lakefs/plugins/diff` if not configured otherwise.

Reviewer: ```suggestion
To enable this feature, you need to install a plugin. You will find the plugin binary in the
[release](https://github.com/treeverse/lakeFS/releases/latest) tarball (versions >= 0.100.0). Put the `delta_diff` binary under `~/.lakefs/plugins/diff` on the machine where lakeFS is running.
```

> lakeFS will look for a binary named `delta_diff` under `~/.lakefs/plugins/diff` if not configured otherwise.

Reviewer 2: If I use a Docker image, do I get it built-in?  Because that sounds like a _really_ good plan :-)  May be worth pulling out what is needed to use with K8s / Docker.

Author: Yes, the docker image is ready to roll

Reviewer: Since most of our users presumably use Docker, maybe first mention that this part is only required when using the binary.

> ### Using Delta Lake table diff

Reviewer: ```suggestion
### Viewing the changes
```

> ### Using the Delta Lake table diff

Reviewer 2: Nit:
```suggestion
### Using Delta Lake table diff
```

{% include toc.html %}

## Beta: Viewing Delta Lake table changes in lakeFS

### Installing the Delta Lake diff plugin

To enable the Delta Lake diff feature, you need to install a plugin. You will find the plugin binary in the
[release](https://github.com/treeverse/lakeFS/releases/latest) tarball (versions >= 0.100.0).

> Put the `delta_diff` binary under `~/.lakefs/plugins/diff` on the machine where lakeFS is running. 

Reviewer 3: really in the user's home folder? it might be worth explaining (briefly) in the text why it's like this and not with the rest of the lakeFS binaries.

> **Notice**: If you're using the lakeFS [docker image](../quickstart/run.md#running-locally-with-docker), the plugin is installed by default.

Reviewer 3: why's it in the Docker image but not the binary distribution?

### Viewing Delta Lake table changes

> Using lakeFS you can compare different versions of Delta tables, view the table operations that have been done since the tables diverged, and the details of those operations.

Reviewer 3: ```suggestion
Using lakeFS you can

* compare different versions of Delta tables
* view the table operations that have been done since the tables diverged and see the details of those operations.
```

Reviewer 3: I would also crop the two subsequent screenshots e.g. 
![CleanShot 2023-03-21 at 10 04 25](https://user-images.githubusercontent.com/3671582/226574470-325db0c4-789e-496c-992c-a2002655e801.png)

I'd also move `Expanding the delete operation, we learn that all movies with a rating < 4 were deleted from the table on the `dev` branch.` to be after the first image to give some separation between them to make it more obvious that they're not just the same image

{% include toc.html %}

## Beta: Viewing Delta Lake table changes in lakeFS 

### Enable the Delta Lake tables diff

The Delta Lake diff functionality is enabled using a binary plugin. This binary plugin can be found as part of the latest lakeFS

> [release](https://github.com/treeverse/lakeFS/releases/latest), and can also be built locally using the `make delta-plugin` command

Author: done

(run from lakeFS's repository root). After getting it, place it under `~/.lakefs/plugins/diff`.  
lakeFS will look for a binary named `delta_diff` under `~/.lakefs/plugins/diff` if not configured otherwise.  

> #### Customizing the Delta Lake diff plugin location

Reviewer 4: I would put it last. Of all the things that I can do with the delta-diff functionality, changing the plugin location is the least interesting of all.

To customize the location of the Delta Lake diff plugin, add the following configurations to the 
[`.lakefs.yaml`](../reference/configuration) file:

```yaml
diff:
  delta:
    plugin: <custom name>

plugins:
  properties:
    <custom name>:
      path: <custom path to the Delta Lake diff plugin binary>
```

For example:

> ```yaml

Reviewer 4: The indentation here and in the previous yaml isn't identical.

So, if you must use multi-cluster writes, you’ll have to allow access from Databricks’ AWS account to lakeFS. 
If you are trying to achieve that, please reach out on Slack and the community will try to assist.

> ### Customizing the Delta Lake diff plugin location

Reviewer: IMO, not important enough: can be part of the configuration reference.

So, if you must use multi-cluster writes, you’ll have to allow access from Databricks’ AWS account to lakeFS. 
If you are trying to achieve that, please reach out on Slack and the community will try to assist.

> ### Customizing the Delta Lake diff plugin location

Reviewer 3: Not sure this belongs under "Best Practices", but regardless, I think it could be condensed significantly and just moved to a note at the end of the `Installing the Delta Lake diff plugin` section, e.g. 

```
You can customize the location of the Delta Lake diff plugin by changing the `plugins.properties.diff.path` configuration in the [`.lakefs.yaml`](../reference/configuration) file. 
```

I don't see the point in documenting `diff.delta.plugin` unless there's a reason why users would want/need to change it (and if there is you probably should expand on it in the docs as it's not clear to me)

Author: True, this was accidentally placed under "Best Practices".
I'll also insert the changes you describe.
Thanks.

> To customize the location of the Delta Lake diff plugin, add the following configurations to the
> [`.lakefs.yaml`](../reference/configuration) file:
> 
> ```yaml
> diff:
>   delta:
>     plugin: <custom name>
> 
> plugins:
>   properties:
>     <custom name>:
>       path: <custom path to the Delta Lake diff plugin binary>
> ```
> 
> For example:
> 
> ```yaml
> diff:
>   delta:
>     plugin: my_delta_diff_plugin
> 
> plugins:
>   properties:
>     my_delta_diff_plugin:
>       path: ~/path/to/delta_plugin
> ```
> 
> It's also possible to change the default location at which lakeFS will look for the `delta_diff` plugin by setting:
> 
> ```yaml
> plugins:
>   default_path: <default location>
> ```
> 
> lakeFS will search for `delta_diff` under the `<default location>/diff` path.

Reviewer 3: ```suggestion

```
